### PR TITLE
feat(analytics): SJIP-954 add analytics page, scatterplot, swarm plot

### PR DESCRIPTION
**Pull request by lflangis**
# FIX

- closes #[TICKET_NUMBER](https://d3b.atlassian.net/browse/SJIP-954)
- closes #[TICKET_NUMBER](https://d3b.atlassian.net/browse/SJIP-955)

### 954
Must Have:
Title
Mouseover tooltip display for the gene, ensembl id, fold change, q-value
Plot headers
Legend
Set the FDR Threshold to the default (q ≤ 0.1). We will not adjust the FDR Threshold based on the facet above the graph since the facets themselves are not must haves for the webinar. Therefore the legend should be as follows:
Down-regulated (q ≤ 0.1)
where the log2(fold change) < 0
Not significant
This means that the dots are q > 0.1
Up-regulated (q ≤ 0.1)
where the log2(fold change)> 0
Actions on plot: zoom in, zoom out, reset (with display of zoom level)
only one gene can be selected at a time until we resolve the layers component with the canvaScatterplot

### 955
Must Have:
Title
Mouseover tooltip display for the sample ID, FPKM value (the sample ID will not have a hyperlink to redirect to the data exploration page – limitation with nivo that we are exploring to resolve)
Swarmplot with values and axes
Boxplot with the min,Q1,median,Q3,Max
Actions on plot, zoom in, zoom out
Legend for control and Trisomy 21 including the count for each

## Screenshot
image
image
image
image

### CODE REVIEW

File: packages/ui/src/components/Charts/ScatterPlot/Canvas/Layers/SelectBoxCanvasLayer.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    ctx.canvas.onmouseup = (event) => {
+        activeMouseCapture = false;
+        origin = { x: 0, y: 0 };
+        handleOnSelect(selectBoxNodes);
+    };
+
+    ctx.canvas.onmousemove = (event) => {
+        const rect = ctx.canvas.getBoundingClientRect();
+        transform = { x: event.clientX - rect.left, y: event.clientY - rect.top };
+        // onSelect(origin, transform);
```
Comment (aperron-ferlab) [inline packages/ui/src/components/Charts/ScatterPlot/Canvas/Layers/SelectBoxCanvasLayer.tsx:104]: 👀

File: packages/ui/src/components/Charts/ScatterPlot/Canvas/index.tsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    Omit<TScatterPlotCanvasExtraProps, 'onAnimationStarted' | 'onAnimationFinished'> & {
+        tooltip?: (value: ScatterPlotTooltipProps<ScatterPlotDatum>) => React.ReactNode;
+    };
+
+/**
+ * ScatterCanvasPlot with zoom feature
+ * No interactive, can be used when you need to display more that 2000 data
+ *
+ * @link https://nivo.rocks/scatterplot/
+ * @returns
```
Comment (aperron-ferlab) [inline packages/ui/src/components/Charts/ScatterPlot/Canvas/index.tsx:42]: returns what?

File: packages/ui/src/components/Charts/ScatterPlot/Svg/Layers/SelectBoxLayer.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+import SelectBox, { TVector2 } from '../../../Core/SelectBox';
+import { IScatterPlotSelectBoxCanvasLayer } from '../../type';
+import { TNodesList } from '../../type';
+import { getSVGRelativeMousePosition } from '../../utils';
+
+/**
+ * ScatterPlot layer
+ * Since zooming/data change re-draw the layer, we must keep track of selected nodes in parent
+ * @returns
```
Comment (aperron-ferlab) [inline packages/ui/src/components/Charts/ScatterPlot/Svg/Layers/SelectBoxLayer.tsx:12]: same here

File: packages/ui/src/components/Charts/ScatterPlot/Svg/index.tsx [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+/**
+ * ScatterPlot with zoom feature
+ * Multiples nodes can be selected through a custom layer ScatterPlotSelectBoxLayer.
+ * Performance threshold is 2000 items
+ *
+ * 1. TODO: FEAT: When zooming with multiples nodes selected, only allow a max zoom only when all nodes can still be visible
+ * 2. TODO: FEAT: add translations
+ *
+ * @link https://nivo.rocks/scatterplot/
+ * @returns
```
Comment (aperron-ferlab) [inline packages/ui/src/components/Charts/ScatterPlot/Svg/index.tsx:56]: :eyes:

File: packages/ui/src/pages/AnalyticsPage/AnalyticsWidget/index.module.css
Comment (aperron-ferlab) [inline packages/ui/src/pages/AnalyticsPage/AnalyticsWidget/index.module.css:1]: empty file

**Review by aperron-ferlab — COMMENTED** [review]
J'allais commenter partout mais peut etre juste checker la JS doc et soit l'enlever ou la completer

File: packages/ui/src/components/Charts/ScatterPlot/Canvas/index.tsx (hunk 2 above)
Comment (lflangis) [reply packages/ui/src/components/Charts/ScatterPlot/Canvas/index.tsx:42]: lol l'autocomplete qui ajoute des returns sans raison

**Review by aperron-ferlab — APPROVED** [review]

**Pull request merged**